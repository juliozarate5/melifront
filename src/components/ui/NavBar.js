import React from 'react';
import PropTypes from 'prop-types';

export default function NavBar({title}) {
    return (
        <nav className="navbar navbar-light bg-light">
        <div className="container-fluid">
            <a className="navbar-brand" href="#">
            <img 
                src="/favicon.svg" 
                alt="" width="30" height="24" 
                className="d-inline-block align-text-top"
            />
            {title}
            </a>
        </div>
        </nav>
    )
}

NavBar.propTypes = {
    title: PropTypes.string.isRequired
}
