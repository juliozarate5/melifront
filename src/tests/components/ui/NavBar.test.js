import React from "react";
import { shallow } from "enzyme";
import NavBar from "../../../components/ui/NavBar";

describe('NavBar tests', () => {

    const title = "title";

    const wrapper = shallow(<NavBar title={title}/>);

    test('showing component successful', () => {

        expect(wrapper).toMatchSnapshot()

    })

    test('must have a with title', () => {

        const a = wrapper.find('a');
        expect( a.text().trim() ).toBe(title);

    })

    test('must have class navbar', () => {

        const nav = wrapper.find('nav');

        const className = nav.prop('className');

        expect(className.includes('navbar')).toBe(true)
    })

})