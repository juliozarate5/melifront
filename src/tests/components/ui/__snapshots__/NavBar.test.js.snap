// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar tests showing component successful 1`] = `
<nav
  className="navbar navbar-light bg-light"
>
  <div
    className="container-fluid"
  >
    <a
      className="navbar-brand"
      href="#"
    >
      <img
        alt=""
        className="d-inline-block align-text-top"
        height="24"
        src="/favicon.svg"
        width="30"
      />
      title
    </a>
  </div>
</nav>
`;
